### ISSUE TITLE
I get paths instead file names after uploading using Windows 10

### ISSUE THREAD

**Issue by daxdax89** [multiplegeorges/vue-cli-plugin-s3-deploy#12]
Here is a screenshot:

http://prntscr.com/kikbl5

**Comment by daxdax89** [comment]
Are you planning on fixing it ? I could really use it. Works well on my macbook but not ok PC. Great work btw. Saves so much time lol



Sent from my Samsung Galaxy smartphone.


-------- Original message --------

**Comment by multiplegeorges** [comment]
@daxdax89 Yeah, but it might take a little while. I have a Windows computer here, but I don't have it set up for Vue development right now.

**Comment by daxdax89** [comment]
Ok thanks. Looking forward to that.



Sent from my Samsung Galaxy smartphone.


-------- Original message --------

**Comment by Pixeleyes** [comment]
` let cwdPrefix = new RegExp(`^${cwd}/${options.assetPath}/`)`

Seems to be a problem here. The C:\ causes problems with the regular expression. I tried to work around it but was still not getting Amazon to see the files properly, I think its because the \ gets escaped to %50 and then amazon does not detect it as a folder.

**Comment by Pixeleyes** [comment]
No problem @daxdax89 sent a mail to the address in your profile.

Main update I made was using node to join the 2 parts of the URL rather than using RegExp. Makes it consistent across platforms.

```
    let cwdPrefix = path.join(cwd, options.assetPath + '/')
    let fileList = getAllFiles(path.join(cwd, options.assetPath))
```
Still need to swap out the slashes though.
`filekey = fileKey.replace("\\", "/")`

**Comment by multiplegeorges** [comment]
Hey guys,

I have opened https://github.com/multiplegeorges/vue-cli-plugin-s3-deploy/pull/18 to address this bug. It should address all the cross-platform path issues.

Could you check out the code at revision 695bb4671580dbbd11839e6e587a81ca863ba83e and verify it locally with `npm link` or `yarn link` ?

Once confirmed, I'll merge it in.

Thanks!

**Comment by Pixeleyes** [comment]
```
(node:44632) UnhandledPromiseRejectionWarning: SyntaxError: Invalid regular expression: /^C:\Users\Pixeleyes\Development\test\dist\/: \ at end of pattern
    at new RegExp (<anonymous>)
    at module.exports
```
You dont need it to be a RegExp as you are doing a string replacement. Thats how I have fixed it locally. Can do a pull request if you want.

**Comment by Pixeleyes** [comment]
Haha love the commit message, fix works 100% on win10

**Comment by multiplegeorges** [comment]
Great, thanks for the verification!

I'll merge it and release an official package very soon. Just working on a couple of other features.

**Comment by crholliday** [comment]
Has this already been released? If so, it is not working as of 2.1.1. Still pushes paths when deployed from WIN 10.

**Comment by Pixeleyes** [comment]
I just confirmed, pulled a new project and 2.1.1 and there is a problem as the windows file names remain folder\file when uploaded to aws when they need to be folder/file.

Adding filekey = fileKey.replace("\\", "/") works. I don't think their should be a case on any platform where someone would want to have a \ in their filename.
